### PR TITLE
[Grouping&Mapping] Swapped to insights client and added props

### PR DESCRIPTION
**Pull request by Guillar1**
Swapped to the external insights client and added some Api options for Design Review and Transformation team.

### CODE REVIEW

**Comment by Guillar1** [comment]
Waiting for insights-client to publish new version.

**Review by bingjiez — APPROVED** [review]

File: packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx [hunk 1]
```diff
@@ -4,20 +4,29 @@
...
+import React from "react";
 import GroupingMapping from "./components/GroupingMapping";
 
 export class GroupingMappingProvider implements UiItemsProvider {
   public readonly id = "GroupingMappingProvider";
 
+  private readonly _accessToken?: AccessToken;
+  private readonly _prefix?: "" | "dev" | "qa";
+
+  constructor(accessToken?: AccessToken, prefix?: "" | "dev" | "qa") {
```
Comment (calebmshafer) [inline packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx:25]: Should this be a prefix or would taking the full URL be better to future proof it?

Comment (Guillar1) [reply packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx:25]: Yes, I originally planned on doing the full URL but imodel-browser-react does not allow this.

Comment (Guillar1) [reply packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx:25]: The new upcoming reports widget will support full url and eventually One Click LCA widget when we get to it.

File: packages/itwin/grouping-mapping-widget/src/widget/components/MappingAction.tsx [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
-import React, { useState } from "react";
-import type { MappingReportingAPI } from "../../api/generated/api";
-import { reportingClientApi } from "../../api/reportingClient";
+import React, { useContext, useState } from "react";
 import ActionPanel from "./ActionPanel";
 import useValidator, { NAME_REQUIREMENTS } from "../hooks/useValidator";
 import { handleError, handleInputChange, WidgetHeader } from "./utils";
 import "./MappingAction.scss";
+import { ApiContext } from "./GroupingMapping";
+import type { Mapping} from "@itwin/insights-client";
```
Comment (arnobmallickbsw) [inline packages/itwin/grouping-mapping-widget/src/widget/components/MappingAction.tsx:12]: style: spacing

**Review by arnobmallickbsw — APPROVED** [review]
looks good to me

File: packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx (hunk 1 above)
Comment (calebmshafer) [reply packages/itwin/grouping-mapping-widget/src/widget/GroupingMappingWidget.tsx:25]: okay, fine by me.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**